### PR TITLE
chore: dart hmac package boilerplate

### PR DESCRIPTION
**Pull request by thorvalld**
- Project skeleton initiated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## 0.0.1
```
Comment (jeanplevesque) [inline CHANGELOG.md]: I don't think you need to add this file. What we tend to do is to only have the breaking changes file instead and use conventional commits to track non-breaking changes.

**Review by jeanplevesque — CHANGES_REQUESTED** [review]

**Comment by jeanplevesque** [comment]
You should also edit the readme to remove the template content and replace it with the information about this recipe.

File: README.md [hunk 2]
```diff
@@ -1,53 +1,18 @@
...
-to install, sample code, etc.}
+This Dart package provides a convenient way to make HTTP requests with HMAC authentication. HMAC (Hash-based Message Authentication Code) authentication is a mechanism for verifying the authenticity of a message using a cryptographic hash function and a secret key.
 
 ## Features
 
-{More details/listing of features of the project}
-
-## Breaking Changes
-
-Please consult [BREAKING_CHANGES.md](BREAKING_CHANGES.md) for more information about version
```
Comment (jeanplevesque) [inline README.md]: The breaking changes section must not be removed.

File: README.md [hunk 3]
```diff
@@ -1,53 +1,18 @@
-# Open Source Project Template
+# HMAC Auth HTTP Requests V1.0.0#
```
Comment (jeanplevesque) [inline README.md]: Don't put version numbers in the title.

**Review by jeanplevesque — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -1,53 +1,78 @@
...
+
+### In your dart code
+
+```
+import 'package:requests_signature_dart/requests_signature_dart.dart';
 
-{More details/listing of features of the project}
+void main() async {
+  // Initialize SignedHMACRequest with your API key and secret
+  final signedRequest = SignedHMACRequest(apiKey, apiSecret);
```
Comment (jeanplevesque) [inline README.md]: The usage described here seem too different from the original's (https://github.com/nventive/RequestsSignature?tab=readme-ov-file#implementing-requests-signature-in-http-client). If you don't want to deal with the readme in this PR, just undo the changes in it. Otherwise, the readme needs to follow the guidelines and the dart recipe must align with the original .NET recipe.

**Review by jeanplevesque — CHANGES_REQUESTED** [review]

**Comment by thorvalld** [comment]
README file reverted to initial template, until we have enough code to document.

**Review by jeanplevesque — APPROVED** [review]

**Review by Soap-141 — APPROVED** [review]
